### ISSUE TITLE
external_storage: mounting Dropbox fails in 8.1

### ISSUE THREAD

**Issue by Webunity** [owncloud/core#17778]
I am unable to mount dropbox to my owncloud 8.1 install.

Precondition:
- Fresh install of owncloud 8.1, running on the root of https://cloud.[mydomain].[ext] (e.g. i am not running on /owncloud, by moving all files to the root of my subdomain)
- PHP 5.6.11 using OPCache and APCu
- Using a real SSL certifcate (e.g. NOT self signed)
- Followed this guide to create dropbox app: https://doc.owncloud.org/server/8.1/admin_manual/configuration_files/external_storage_configuration_gui.html
- As admin, enabled the external_storage app
- Entered the credentials into external_storage;
  - Map; Dropbox
  - Provider; Dropbox
  - app key: [my key from dropbox]
  - app secret: [my secret from dropbox]
  - Click authorize

Shows RED square even though:
- I get a mail from dropbox that i've connected an app
- I see 1/100 users in my app console
- I see a mount.js file in my data directory with contents:
```
{
    "user": {
        "all": {
            "\/$user\/files\/Dropbox": {
                "id": 1,
                "class": "\\OC\\Files\\Storage\\Dropbox",
                "options": {
                    "configured": "false",
                    "app_key": "...",
                    "app_secret": "...",
                    "token": "N8aLsCRHpzpaq8nW",
                    "token_secret": "R0hF4mLbB4h5A3GL"
                },
                "priority": 100
            }
        }
    }
}
```

So it seems dropbox authenticated succesfully, however OwnCloud is not working.. I tried with URLs
- https://localhost/index.php/settings/personal
- https://localhost/index.php/settings/admin

As well as
- https://cloud.[mydomain].[ext]/index.php/settings/personal
- https://cloud.[mydomain].[ext]/index.php/settings/admin

Even multiple combinations of fresh installs but nothing seems to work.

The PUT results in a 403:
```
Remote Address:[ip]
Request URL:https://cloud.[mydomain].[ext]/index.php/apps/files_external/globalstorages/1
Request Method:PUT
Status Code:403 Forbidden
```

Additionally; I am also unable to remove this failed storage mount (HTTP DELETE gives also 403 error).

**Comment by Webunity** [comment]
Ok, problem was related to my upgrade to Apache 2.4.x

Looking at the log files, i saw a lot of these errors 
```
AH01797: client denied by server configuration: /.../private_html/index.php, referer: https://cloud.[domain].[ext]/index.php/settings/admin
```

That seemed to be caused by my upgrade to Apache 2.4.x which has breaking changes to the authorization scheme. It replaces the "Order deny,allow + Deny from all" and "Order allow,deny + Allow from all" with respectively "Require all denied" and "Require all granted".
More information here: http://httpd.apache.org/docs/2.4/upgrading.html#run-time

After replacing all the configuration files, i saw a new error popping up:
```
AH01630: client denied by server configuration: /.../private_html/index.php, referer: https://cloud.[domain].[ext]/index.php/settings/admin
```

This one was fixed by looking at http-directories.conf; The setting
```
<Directory />
    AllowOverride none
    Require all denied
</Directory>
```

Basically disables all access to every directory in your webserver.

So in order to re-enable the acces, for your ALLOWED directories (e.g. your /home folders), you should add something like this:
```
<Directory /home>
		# ...

    Require all granted

		# ...
</Directory>
```

Then i could again click "Authorize" and all worked fine, i got a green light next to my DropBox.

Hope this helps someone.
